Code/Framework/AzCore/AzCore/Debug/Budget.h

```diff
@@ -62,12 +62,16 @@ namespace AZ::Debug
 //
 // Anywhere the budget is used, the budget must be declared (either in a header or in the source file itself)
 // AZ_DECLARE_BUDGET(AzCore);
-#define AZ_DEFINE_BUDGET(name)                                                                                                             \
-    ::AZ::Debug::Budget* AZ_BUDGET_GETTER(name)()                                                                                          \
-    {                                                                                                                                      \
-        constexpr static uint32_t crc = AZ_CRC_CE(#name);                                                                                  \
-        static ::AZ::Debug::Budget* budget = ::AZ::Debug::BudgetTracker::GetBudgetFromEnvironment(#name, crc);                             \
-        return budget;                                                                                                                     \
+#define AZ_DEFINE_BUDGET(name)                                                          \
+    ::AZ::Debug::Budget* AZ_BUDGET_GETTER(name)()                                       \
+    {                                                                                   \
+        static ::AZ::Debug::Budget* budget = nullptr;                                   \
```

Comment:
I see you went back to raw pointers (and a registration to reset it) instead of a `weak_ptr`. I am ok with this, but curious about the reason

Reply:
Ran into more issues with freeing the weak count.  While using the stateless allocator did solve the issue on module unload, it did not completely fix on program exit, specifically when the same budget was accessed across DLL boundaries.  Can thank the Linux test runner for properly catching the exit crash as it wasn't on Windows, which according to the debugger wasn't consistently happening either.